### PR TITLE
Add BitKeep Wallet 

### PR DESCRIPTION
**Pull request by bitkeepwallet**

### CODE REVIEW

File: packages/bitkeep/src/adapter.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+            this.emit('error', error);
+            throw error;
+        }
+    }
+
+    private _messaged = (event: MessageEvent) => {
+        const data = event.data;
+        if (data && data.origin === 'bitkeep_internal' && data.type === 'lockStatusChanged' && !data.payload) {
+            this._disconnected();
+        }
```
Comment (jordansexton) [inline packages/bitkeep/src/adapter.ts:152]: This appears to be copied from MathWallet. Unless BitKeep works by using a popup window in the same way as MathWallet, I would recommend basing this off a different adapter. See https://github.com/solana-labs/wallet-adapter/blob/master/packages/solflare/src/adapter.ts for how Solflare's adapter is structured.

File: packages/wallets/src/wallets.ts [hunk 2]
```diff
@@ -17,6 +18,7 @@ export enum WalletName {
     Coin98 = 'Coin98',
     Ledger = 'Ledger',
     MathWallet = 'MathWallet',
+    BitKeep = 'BitKeep',
```
Comment (jordansexton) [inline packages/wallets/src/wallets.ts:21]: This enum is alphabetical.

File: packages/wallets/src/wallets.ts [hunk 3]
```diff
@@ -70,6 +72,13 @@ export const getMathWallet = (config?: MathWalletWalletAdapterConfig): Wallet =>
     adapter: () => new MathWalletWalletAdapter(config),
 });
 
+export const getBitKeepWallet = (config?: BitKeepWalletAdapterConfig): Wallet => ({
+    name: WalletName.BitKeep,
+    url: 'https://bitkeep.com',
+    icon: `${ICONS_URL}/bitkeep.svg`,
+    adapter: () => new BitKeepWalletAdapter(config),
+});
+
```
Comment (jordansexton) [inline packages/wallets/src/wallets.ts:81]: These functions are alphabetical.

**Review by jordansexton — CHANGES_REQUESTED** [review]
Thanks for this contribution. The MathWallet adapter isn't an ideal example for what adapter to base this on. I recommend checking out how Solflare's adapter is structured.

**Comment by jordansexton** [comment]
Can you point me to documentation for Bitkeep's support for Solana?

**Comment by jordansexton** [comment]
Can you point me to documentation for BitKeep's support for Solana?

File: packages/bitkeep/src/adapter.ts [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+interface BitKeep {
+    isBitKeep?: boolean;
+    getAccount(): Promise<string>;
+    signTransaction(transaction: Transaction): Promise<Transaction>;
+    signAllTransactions(transactions: Transaction[]): Promise<Transaction[]>;
+}
+
+interface BitKeepWindow extends Window {
+    solana?: BitKeep;
```
[the remark this answers is not in the document]
Comment (bitkeepwallet) [reply packages/bitkeep/src/adapter.ts:22]: BitKeep is a app wallet. If use `window.bitkeep,` then the SOL Dapps couldn't connect user's wallet.

File: packages/bitkeep/src/adapter.ts [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            this._connecting = true;
+
+            const wallet = typeof window !== 'undefined' && window.solana;
+            if (!wallet) throw new WalletNotFoundError();
+            if (!wallet.isBitKeep) throw new WalletNotInstalledError();
+
+            // @TODO: handle if popup is blocked
```
[the remark this answers is not in the document]
Comment (bitkeepwallet) [reply packages/bitkeep/src/adapter.ts:75]: No, BitKeep is an mobile app.

**Comment by bitkeepwallet** [comment]
Document pace

> Can you point me to documentation for BitKeep's support for Solana?

https://docs.bitkeep.com/dapp-development/connect/solana-connect

In order to be compatible with the existing Sol DApps, we directly adopted a scheme similar to MathWallet, which allows Dapps to connect directly in BitKeep and initiate transactions.

File: packages/bitkeep/src/adapter.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (jordansexton) [reply packages/bitkeep/src/adapter.ts:22]: Can you clarify why `window.bitkeep` wouldn't work? All of the adapters in this library besides MathWallet and Phantom use `window.somethingOtherThanSolana` and they connect, and we are planning on phasing all use of `window.solana` out.

File: packages/bitkeep/src/adapter.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (jordansexton) [reply packages/bitkeep/src/adapter.ts:75]: This should be removed if Bitkeep doesn't use a popup window.

File: packages/wallets/src/wallets.ts (hunk 2 above)
Comment (bitkeepwallet) [reply packages/wallets/src/wallets.ts:21]: Ok, We will fix

File: packages/bitkeep/src/adapter.ts (hunk 1 above)
Comment (bitkeepwallet) [reply packages/bitkeep/src/adapter.ts:152]: Ok, we will update as Solfare's adapter.

**Pull request closed without merging**